framework/src/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -49,12 +50,154 @@ public static ContractsUtils getInstance(
     }
 
     /**
+     * Represents a pre- or postcondition that must be verified by
+     * {@code BaseTypeVisitor} or one of its subclasses. Automatically
+     * extracted from annotations with meta-annotations
+     * {@code @PreconditionAnnotation} or {@code @PostconditionAnnotation},
+     * such as {@code @RequiresNonNull} and {@code @EnsuresNonNull}.
+     * Can also be generated, such as in
+     * {@code LockVisitor.generatePreconditionsBasedOnGuards}.
+     */
+    public static class PreOrPostcondition {
```

Comment:
This class is great! It made looking through the code so much easier compared to the previous Pair!